### PR TITLE
Basic Reason support

### PR DESCRIPTION
**Pull request by vasilisp**
This PR provides very rough and mostly untested Reason support.

I added a `-reason` flag to our compiler wrappers. It is implemented by calling `refmt` as a preprocessor.

There is a new `basic.reason` template that uses Reason syntax.

### CODE REVIEW

**Comment by vasilisp** [comment]
There is a problem with type-checking client values (and probably injections). It is because the new preprocessing pipeline introduces unpredictable intermediate filenames, impacting our auto-generated names for client fragments. To be fixed tomorrow.

File: src/tools/utils.ml [hunk 1]
```diff
@@ -362,6 +364,12 @@ let get_ppopts ~impl_intf file =
     type_opt impl_intf file @ !ppopt
 
 let preprocess_opt ?(ocaml = false) ?kind opts =
+  let refmt () =
+    if !use_refmt then
+      [ "-pp"    ; "refmt -parse re -print ml" ]
```
Comment (Drup) [inline src/tools/utils.ml]: Should we try to provide a clean failure when reason is not installed ?

**Comment by Drup** [comment]
Nice work!

**Comment by vasilisp** [comment]
@Drup, does e706f1d0602f849256507030ea627430525f0fa5 look line a sane approach?

**Comment by vasilisp** [comment]
See #504.

**Pull request closed without merging**